registry/registry.go

```diff
@@ -319,7 +324,17 @@ func (r *RouteRegistry) pruneStaleDroplets() {
 
 	r.byURI.EachNodeWithPool(func(t *container.Trie) {
 		endpoints := t.Pool.PruneEndpoints()
-		t.Snip()
+		if r.EmptyPoolResponseCode503 {
+			if t.Pool.EmptyPoolTimeout == 0 {
+				t.Pool.EmptyPoolTimeout = 30 * time.Second
+			}
+			if time.Since(t.Pool.LastUpdated()) > t.Pool.EmptyPoolTimeout {
+				t.Snip()
+			}
+		} else {
+			t.Snip()
+		}
+
```

Comment:
```suggestion
		if r.EmptyPoolResponseCode503 && t.Pool.EmptyPoolTimeout > 0 {
			if time.Since(t.Pool.LastUpdated()) > t.Pool.EmptyPoolTimeout {
				t.Snip()
			}
		} else {
			t.Snip()
		}

```